services/api/src/models/user.js

```diff
@@ -16,6 +16,22 @@ schema.virtual('password').set(function (password) {
   this._password = password;
 });
 
+schema.methods.addAuthToken = function ({ exp, jti, iat }, ctx) {
+  // filter out any expired tokens + prevent that we are not adding a duplicate
+  const authTokens = (this.authTokens || []).filter((token) => token.jti !== jti && token.exp > Date.now());
```

Comment:
this `||` should be redundant here as it will default to an empty array due to the schema

Reply:
another question...should there be any reason to access tokens after they've expired? also maybe adding a new token isn't the place to do this...what about a pre save hook?

Reply:
well it could only trigger a save if there are tokens to be filtered out. or perhaps expired tokens shouldn't be filtered out at all but only overwritten when logging in or removed when logging out/

Reply:
whats the use case for them if they are expired? just trying to prevent an ever growing array

Reply:
I'm not sure what you mean by "use case".. i'm just trying to figure out where/how to expire them... for example one possibility is when a new token is issues on login any other tokens with the same userAgent could be filtered out

Reply:
kk